apps/server/src/modules/board/uc/media-board/media-element.uc.ts

```diff
@@ -44,4 +53,67 @@ export class MediaElementUc {
 			throw new FeatureDisabledLoggableException('FEATURE_MEDIA_SHELF_ENABLED');
 		}
 	}
+
+	public async createElement(
+		userId: EntityId,
+		schoolExternalToolId: EntityId,
+		lineId: EntityId,
+		position: number
+	): Promise<MediaExternalToolElement> {
+		this.checkFeatureEnabled();
+
+		const line: MediaLine = await this.mediaLineService.findById(lineId);
+
+		const user: User = await this.checkUsersPermissions(userId, line);
+
+		const mediaBoard: MediaBoard = await this.mediaBoardService.findByLine(line);
+
+		const schoolExternalTool: SchoolExternalToolWithId = await this.schoolExternalToolService.findById(
+			schoolExternalToolId
+		);
+
+		await this.checkElementExistsAlreadyOnBoardAndThrow(mediaBoard, schoolExternalTool);
+
+		const createdContexExternalTool: ContextExternalToolWithId =
+			await this.mediaElementService.createContextExternalToolForMediaBoard(user, schoolExternalTool, mediaBoard);
+
+		const createdElement: AnyMediaContentElementDo = await this.mediaElementService.createExternalToolElement(
+			line,
+			position,
+			createdContexExternalTool
+		);
+
+		return createdElement;
+	}
+
+	private async checkElementExistsAlreadyOnBoardAndThrow(
+		mediaBoard: MediaBoard,
+		schoolExternalTool: SchoolExternalToolWithId
+	): Promise<void> {
+		const exists = await this.mediaElementService.checkElementExists(mediaBoard, schoolExternalTool);
+
+		if (exists) {
+			throw new MediaBoardElementAlreadyExistsLoggableException(mediaBoard.id, schoolExternalTool.id);
+		}
+	}
+
+	public async deleteElement(userId: EntityId, elementId: EntityId): Promise<void> {
+		this.checkFeatureEnabled();
+
+		const element: AnyMediaContentElementDo = await this.mediaElementService.findById(elementId);
+
+		await this.checkUsersPermissions(userId, element);
+
+		await this.mediaElementService.delete(element);
+	}
+
+	private async checkUsersPermissions<T extends AnyMediaBoardDo>(userId: EntityId, mediaBoard: T): Promise<User> {
```

Comment:
I would not group the permission check in one function since this leads to unintentional reuse of the function. In this case it works atm, but when you have different permissions that you want to check or if the operation is a read operation instead of a write operation, then you wouldn't know from just reading the function name. E.g. you would assume that "every function here uses it so it must work for my new use case, too" but then the operation type is messed up or you check the wrong permission.